### PR TITLE
VOA-2349: Refactor Cr03submission

### PR DESCRIPTION
**Pull request by valydia**

### CODE REVIEW

File: app/controllers/ConfirmationController.scala [hunk 1]
```diff
@@ -101,11 +101,14 @@ class ConfirmationController @Inject()(appConfig: FrontendAppConfig,
...
   }
 
-  private def getCr03(reportStatus: ReportStatus): Option[Cr03Submission] = {
+  private def getCr01Cr03(reportStatus: ReportStatus): Option[Cr01Cr03Submission] = {
     reportStatus.report
       .map(_.value)
-      .filter(x => x.get("type").map {case x: JsString => x.value == "Cr03Submission"}.getOrElse(false))
-      .flatMap(x => x.get("submission")).flatMap(x => Cr03Submission.format.reads(x).asOpt)
+      .filter(x => x.get("type").exists {
+        case x: JsString => x.value == "Cr01Cr03Submission"
```
Comment (valydia) [inline app/controllers/ConfirmationController.scala:108]: ⚠️ Breaking changes

File: app/journey/RemovalReasonType.scala [hunk 2]
```diff
@@ -22,9 +22,9 @@ sealed trait RemovalReasonType
 
 case object Demolition extends RemovalReasonType
 case object Disrepair extends RemovalReasonType
+case object Derelict extends RemovalReasonType
```
Comment (valydia) [inline app/journey/RemovalReasonType.scala:25]: Depends on https://github.com/hmrc/voa-bar/pull/107/files#diff-382fb5e4fe8036adfa7af45318d296e8R32

File: app/controllers/ConfirmationController.scala (hunk 1 above)
Comment (rhubner) [inline app/controllers/ConfirmationController.scala:108]: So backend and frontend will always use `Cr01Cr03Submission` and backend have compatibility for previous `Cr03Submission` which can me in database.

**Review by rhubner — APPROVED** [review]

File: app/controllers/ConfirmationController.scala (hunk 1 above)
Comment (valydia) [reply app/controllers/ConfirmationController.scala:108]: Yes - this is the line for backward compatibility https://github.com/hmrc/voa-bar/blob/master/app/uk/gov/hmrc/voabar/services/WebBarsService.scala#L74

**Pull request merged**